distributed/utils_test.py

```diff
@@ -1004,9 +1005,26 @@ async def coro():
                             task.cancel()
                             while not task.cancelled():
                                 await asyncio.sleep(0.01)
+
+                            # Remove as much of the traceback as possible; it's
+                            # uninteresting boilerplate from utils_test and asyncio and
+                            # not from the code being tested.
                             raise TimeoutError(
-                                f"Test timeout after {timeout}s.\n{buffer.getvalue()}"
-                            ) from e
+                                f"Test timeout after {timeout}s.\n"
+                                "========== Test stack trace starts here ==========\n"
+                                f"{buffer.getvalue()}"
+                            ) from None
+
+                        except Exception:
+                            if cluster_dump_directory:
```

Comment:
```suggestion
                            if cluster_dump_directory and not hasattr(test_func, "xfail"):
```
Seems like maybe this would work based on https://stackoverflow.com/a/31242341/17100540?